### ISSUE TITLE
Updating variable product short description removed the product attributes

### ISSUE THREAD

**Issue by reginabally** [woocommerce/woocommerce-android#3833]
**Describe the bug**

It was reported in 3903610-zd-woothemes that editing the short description of a variable product through the app will cause the product attributes to be removed.

**To Reproduce**
1. Go to Products tab, select a variable product to edit
2. Change the "Short description" of the product and tap "UPDATE"
3. Tap three dots > View product on the store and notice the product variations are removed
4. Go to WP Admin > WooCommerce > Products > All Products, select the variable product that was edited in the app.
5. Go to the Attributes section, notice all of the product attributes are removed.

**Screen recording**

https://user-images.githubusercontent.com/40906847/114498842-6e353280-9c57-11eb-8d7b-2a6dbeb703ea.mp4

**Expected behavior**

**Isolating the problem (mark completed items with an [x]):**
- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: OPPO Reno 4
- Android version: 11
- WooCommerce Android version: 6.4-rc-1

**WordPress Environment**
<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://woo20200908.mystagingwebsite.com
Site address (URL): https://woo20200908.mystagingwebsite.com
WC Version: 5.1.0
REST API Version: ✔ 5.1.0
WC Blocks Version: ✔ 4.4.3
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 2.0.2
Log Directory Writable: ✔
WP Version: 5.7
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.3.27
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.75.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.24-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.1.0
WC Database Prefix: wp_
Total Database Size: 6.70MB
Database Data Size: 4.72MB
Database Index Size: 1.98MB
wp_woocommerce_sessions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mo2f_network_blocked_ips: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_network_email_sent_audit: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_network_transactions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_network_whitelisted_ips: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_user_details: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo2f_user_login_info: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 0.30MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.22MB + Index: 0.14MB + Engine InnoDB
wp_posts: Data: 0.14MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfblockediplog: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfblocks7: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfconfig: Data: 0.47MB + Index: 0.00MB + Engine InnoDB
wp_wfcrawlers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilechanges: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilemods: Data: 0.42MB + Index: 0.00MB + Engine InnoDB
wp_wfhits: Data: 0.06MB + Index: 0.05MB + Engine InnoDB
wp_wfhoover: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfknownfilelist: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_wflivetraffichuman: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wflocs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wflogins: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wfls_2fa_secrets: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfls_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfnotifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfpendingissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfreversecache: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfsnipcache: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfstatus: Data: 0.09MB + Index: 0.03MB + Engine InnoDB
wp_wftrafficrates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_attack_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_backup_report: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_files_scan: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpns_ip_rate_details: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_malware_hash_file: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wpns_malware_scan_report: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_malware_scan_report_details: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpns_malware_skip_files: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

attachment: 59
page: 10
post: 6
product: 31
product_variation: 23
revision: 10
seedprod: 1
shop_coupon: 1
shop_order: 25
shop_order_refund: 3
wc_order_status: 7

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (2) ###

Jetpack by WordPress.com: by Automattic – 9.6
WooCommerce: by Automattic – 5.1.0

### Inactive Plugins (15) ###

Akismet Anti-Spam: by Automattic – 4.1.9
All in One Product Quantity for WooCommerce: by WPWhale – 3.7.2
Classic Editor: by WordPress Contributors – 1.6
Coming Soon Page, Maintenance Mode & Landing Pages by SeedProd: by SeedProd – 6.2.1
Custom Order Status for WooCommerce: by Tyche Softwares – 2.0.2
Enable jQuery Migrate Helper: by The WordPress Team – 1.3.0
Extra Product Options (Product Addons) for WooCommerce: by ThemeHigh – 3.0.2
miniOrange 2 Factor Authentication: by miniOrange – 5.4.33
Under Construction: by WebFactory Ltd – 3.88
WooCommerce Order Status Manager: by SkyVerge – 1.13.0
WooCommerce Product Add-ons: by WooCommerce – 3.3.1
WooCommerce Shipment Tracking: by WooCommerce – 1.6.26
WooCommerce Stripe Gateway: by WooCommerce – 5.1.0
Wordfence Security: by Wordfence – 7.5.2
XML-RPC Validator: by Danilo E – 0.1

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

API Enabled: –
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #104 - /
Cart: #105 - /cart/
Checkout: #106 - /checkout/
My account: #107 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 3.5.1
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 14
Oldest: 2021-04-06 00:38:24 +0000
Newest: 2021-04-13 04:48:24 +0000

Pending: 1
Oldest: 2021-04-13 04:56:03 +0000
Newest: 2021-04-13 04:56:03 +0000


### Status report information ###

Generated at: 2021-04-13 04:55:17 +00:00
`
```
</details>

**Comment by anitaa1990** [comment]
I've been looking into this issue and it turns out there are two separate issues. 
1. We seem to be sending the `attributes` object in the API request when updating the product, even though the attributes section has not been updated. **We should not be sending the `attributes` object in the API request when it has not changed.**
2. When we are sending the `attributes` object in the API request when updating the product, we fail to add the `variation` field. This flag is what determines whether the attributes can be used for variations. **We accidentally fail to send this field so the attributes/variations disappear from the product store.**

**Solution**
I have opened a PR [here](https://github.com/woocommerce/woocommerce-android/pull/3834) to fix Issue 2. So the `variation` field in now included in the API request when updating a product. This will ensure the attributes/variations will not disappear from the store. 

In order to fix Issue 1, we need to include the `position` field in the `ProductAttribute` model in FluxC and update the `ProductAttribute` in Woo to include those changes. I tried to fix that but it seems to be a lot more changes than I expected and I'm not sure why this was excluded in the first place. and since work is being done at the moment for variations by @ThomazFB and @nbradbury I would like to leave it up to them to decide if this field needs to be included. 

Since this issue feels important I thought it best to target the release branch instead of `develop`.
